### ISSUE TITLE
How to compile x264 for use in an iPhone application.

### ISSUE THREAD

**Issue by fxf601244325** [lajos/iFrameExtractor#1]
Hi lajos,

  iFrameExtractor made me grow a lot.

  I'm learning How to compile x264 for use in an iPhone application.

  My version is 3.2.5 xcode，and I get x264-armv6 with https://github.com/gabriel/ffmpeg-iphone-build/issues#issue/1.

 ./configure --disable-doc --disable-ffmpeg --disable-ffplay --disable-ffserver --enable-cross-compile --arch=arm --target-os=darwin --cc=/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/gcc --as='gas-preprocessor/gas-preprocessor.pl /Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/gcc' --sysroot=/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS4.2.sdk --cpu=arm1176jzf-s --extra-cflags='-arch armv6' --extra-ldflags='-arch armv6' 

Obtained config.err at compile, and My libzvcodes.a size 19.7M, that I have not added x264.

  I have something else?

  sory my poor English.

Thanks.

**Comment by lajos** [comment]
Hi-

libx264 is separate from ffmpeg. It's not included in this project and I don't have any experience with compiling it for iphone.

-lajos

**Comment by fxf601244325** [comment]
Hi-

I have successfully compiled together x264 and ffmpeg.

......................

Checked out revision 25924.
Configure options: --enable-gpl --enable-postproc --enable-swscale --enable-avfilter --enable-libx264 --extra-ldflags=-L/Users/fxf/Desktop/gabriel-ffmpeg-iphone-build-b961668/x264-armv6/dist/lib --extra-cflags=-I/Users/fxf/Desktop/gabriel-ffmpeg-iphone-build-b961668/x264-armv6/dist/include
install prefix            dist
source path               /Users/fxf/Desktop/gabriel-ffmpeg-iphone-build-b961668/ffmpeg-armv6
C compiler                /Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/gcc-4.2
ARCH                      arm (arm1176jzf-s)
big-endian                no
runtime cpu detection     no
ARMv5TE enabled           no
ARMv6 enabled             no
ARMv6T2 enabled           no
ARM VFP enabled           no
IWMMXT enabled            no
NEON enabled              no
debug symbols             yes
strip symbols             yes
optimizations             yes
static                    yes
shared                    no
postprocessing support    yes
new filter support        yes
network support           yes
threading support         pthreads
SDL support               no
Sun medialib support      no
AVISynth enabled          no
frei0r enabled            no
libdc1394 support         no
libdirac enabled          no
libfaac enabled           no
libgsm enabled            no
libmp3lame enabled        no
libnut enabled            no
libopencore-amrnb support no
libopencore-amrwb support no
libopencv support         no
libopenjpeg enabled       no
librtmp enabled           no
libschroedinger enabled   no
libspeex enabled          no
libtheora enabled         no
libvorbis enabled         no
libvpx enabled            no
libx264 enabled           yes
libxavs enabled           no
libxvid enabled           no
zlib enabled              yes
bzlib enabled             no
.......................................

but
       2010-12-10 11:09:15.542 out_h264[1988:307] /var/mobile/Applications/35571783-    CB9A-4264-84F8-4D759F1F578E/Documents/out.mp4
      Video encoding
      [libx264 @ 0x1021200] broken ffmpeg default settings detected
      [libx264 @ 0x1021200] use an encoding preset (e.g. -vpre medium)
      [libx264 @ 0x1021200] preset usage: -vpre <speed> -vpre <profile>
      [libx264 @ 0x1021200] speed presets are listed in x264 --help
      [libx264 @ 0x1021200] profile is optional; x264 defaults to high
      could not open codec

..........................
I got the link http://forum.doom9.org/showthread.php?t=148750
............
      ffmpeg's defaults are broken and they override x264's encoding settings with   completely broken ones. There doesn't seem to be much effort among ffmpeg   developers to get this fixed, or for that matter, accept patches that fix them.

```
    As a result, x264 will refuse to encode if it detects that you are using broken ffmpeg default settings and tells you to use -vpre instead with one of the included presets.
```

...............
I will continue to try to resolve it.

Thanks.

-fxf601244325
